### DISCUSSION TITLE
I need some help on how to use the different RuleContextModes values

### DISCUSSION THREAD

**Discussion by GreatBarrier86** [MarimerLLC/csla#2924] [Questions]
I could use some explanation for how the different `RuleContextModes` values work.

`CheckRules` seems to mean someone manually called `CheckRules`/`CheckRules(MyProperty)`
`CheckObjectRules` seems to mean someone manually called `CheckObjectRules()`

I'm unclear on how `PropertyChanged` differs from `AsAffectedProperty`?  Does `PropertyChanged` mean the rule fired because that property associated with the rule had changed and `AsAffectedProperty` means that the rule ran since that property was dependent on the one that did change?

![image](https://user-images.githubusercontent.com/29482393/166989274-2e0f93ff-6c78-4282-9ecf-f1969eee664d.png)

**Comment by TheCakeMonster** [comment]
I don't know the answer for sure and definite, but it certainly seems to me that you are right in what you have surmised. I would read it exactly as you have, as your understanding of it seems to be reinforced by the comments. AsAffectedProperty will be to do with rules where an update to one property - the primary property to which the rule is attached - are marked as also affecting one or more other properties.

**Reply by GreatBarrier86** [reply]
@rockfordlhotka, do @TheCakeMonster and I basically have this correct? If you have a moment, I'd appreciate any other feedback you might have on it.

**Reply by rockfordlhotka** [reply]
The expert is @jonnybee, though his career has moved him off CSLA.

I think you've summarized it as I understand it. Have you looked at the rules sample apps? Jonny wrote them to illustrate how this works.
